docs/learn/smart-contract-developers/create-nft-collection.md

```diff
@@ -0,0 +1,184 @@
+
+## Setup
+
+Make sure you have the following dependencies installed:
+
+- [`ethers.js v6`](https://github.com/ethers-io/ethers.js/) (alternatively you can use [`web3.js`](https://github.com/web3/web3.js))
+- [`@lukso/lsp-smart-contracts`](https://github.com/lukso-network/lsp-smart-contracts/)
+
+```bash
+npm install ethers @lukso/lsp-smart-contracts
+```
+
+## Create the Smart Contracts
+
+When creating smart contracts representing digital assets on LUKSO, you will need to specify the token type and data keys for the 📄 [LSP4 Digital Asset Metadata](../../standards/tokens/LSP4-Digital-Asset-Metadata) that will be stored in the 🗂️ [ERC725Y](../../standards/lsp-background/erc725.md#erc725y-generic-data-keyvalue-store) storage of the Digital Asset. There are three different token types:
+
+- `0` = Token
+- `1` = NFT
+- `2` = Collection
+
+```solidity title="contracts/TokenTypes.sol"
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.4;
+
+bytes32 constant _LSP4_TOKEN_TYPE_DATA_KEY = 0xe0261fa95db2eb3b5439bd033cda66d56b96f92f243a8228fd87550ed7bdfdb3; // kecca256 hash of the word `LSP4TokenType`
+
+enum TokenType {
+    TOKEN,
+    NFT,
+    COLLECTION
+}
+```
+
+After defining the type of the asset and its 🗂️ [ERC725 data key](../../standards/lsp-background/erc725.md#erc725y-generic-data-keyvalue-store) you can create a custom 🌄 [LSP8 Identfiable Digital Asset Collection](../../standards/tokens/LSP8-Identifiable-Digital-Asset.md) that extends [LSP8Mintable](../../contracts/contracts/LSP8IdentifiableDigitalAsset/presets/LSP8Mintable.md) so that new assets can be created within the smart contract.
+
+```solidity title="contracts/MyNFTCollection.sol"
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.4;
+
+// modules
+import {
+    LSP8Mintable
+} from "@lukso/lsp-smart-contracts/contracts/LSP8IdentifiableDigitalAsset/presets/LSP8Mintable.sol";
+
+// constants
+import {
+    _LSP8_TOKENID_TYPE_NUMBER
+} from "@lukso/lsp-smart-contracts/contracts/LSP8IdentifiableDigitalAsset/LSP8Constants.sol";
+import {_LSP4_TOKEN_TYPE_DATA_KEY, TokenType} from "../TokenTypes.sol";
+
+contract BasicNFTCollection is LSP8Mintable {
+    constructor(
+        string memory nftCollectionName,
+        string memory nftCollectionSymbol,
+        address contractOwner
+    )
+        LSP8Mintable(
+            nftCollectionName,        // NFT collection name
+            nftCollectionSymbol,      // NFT collection symbol
+            contractOwner,            // owner of the NFT contract (the address that controls it, sets metadata, can transfer the ownership of the contract)
+            _LSP8_TOKENID_TYPE_NUMBER // type of NFT/ tokenIds
+        )
+    {
+        // set token type
+        _setData(_LSP4_TOKEN_TYPE_DATA_KEY, abi.encode(TokenType.COLLECTION));
+    }
+}
+```
+
+Next you define the deployment script.
+
+<Tabs groupId="web3-lib">
+  <TabItem value="ethersjs" label="ethers.js">
+
+<!-- prettier-ignore-start -->
+```js title="deployLSP8Collection.ts"
```

Comment:
same this file is nowhere to be found in the related repo

<img width="350" alt="image" src="https://github.com/lukso-network/docs/assets/477945/1eea5a97-4bfb-4be3-888d-6490030c9f3a">

https://github.com/CJ42/LUKSO-Hardhat-template